spring-integration-stomp/src/main/java/org/springframework/integration/stomp/AbstractStompSessionManager.java

```diff
@@ -127,15 +148,39 @@ public void onFailure(Throwable e) {
 
 			@Override
 			public void onSuccess(StompSession stompSession) {
-				stompSession.setAutoReceipt(autoReceipt);
+				AbstractStompSessionManager.this.connecting = false;
+				stompSession.setAutoReceipt(isAutoReceiptEnabled());
 				AbstractStompSessionManager.this.connected = true;
+				AbstractStompSessionManager.this.reconnectFuture = null;
```

Comment:
Well, we might, but that may be in conflict with the `StompCommand.CONNECTED` STOMP Frame from server in response to `StompCommand.CONNECT` Frame.

We can revise that one more time, but I don't think that it should be part of this PR.
WDYT?